### PR TITLE
Add read load part with unit test.

### PR DESCRIPTION
**Pull request by killian-scalian**

### CODE REVIEW

File: src/antares/model/area.py [hunk 1]
```diff
@@ -372,3 +372,6 @@ def create_hydro(
         hydro = self._area_service.create_hydro(self.id, properties, matrices)
         self._hydro = hydro
         return hydro
+
+    def read_load(self) -> Load:
```
Comment (MartinBelthle) [inline src/antares/model/area.py:376]: There's already a method named `get_load_matrix` inside area you should use this and just put your code inside the local implementation

File: src/antares/service/local_services/area_local.py [hunk 2]
```diff
@@ -334,3 +335,33 @@ def read_areas(self) -> List[Area]:
...
+                    sep="\t",
+                    header=None,
+                    index_col=None,
+                    encoding="utf-8",
+                )
+            else:
+                _time_series = pd.DataFrame()
+        else:
+            _time_series = time_series if time_series is not None else pd.DataFrame([])
+            _write_file(file_path, _time_series)
```
Comment (MartinBelthle) [inline src/antares/service/local_services/area_local.py:362]: Why would you want to write a file inside a read method ? You shouldn't do this.

Also you could use the `df_read` method introduced recently to centralize the code relative to dataframes.
You could add the if else with the file_size inside this method  as it's useful

File: src/antares/service/local_services/area_local.py [hunk 3]
```diff
@@ -334,3 +335,33 @@ def read_areas(self) -> List[Area]:
...
         return areas
+
+    def _read_timeseries(self, ts_file_type: TimeSeriesFileType, study_path: Path, area_id: Optional[str] = None, group_id: Optional[str] = None,constraint_id: Optional[str] = None, time_series: Optional[pd.DataFrame] = None,) -> pd.DataFrame:
+        def _write_file(_file_path: Path, _time_series: pd.DataFrame) -> None:
+            _file_path.parent.mkdir(parents=True, exist_ok=True)
+            _time_series.to_csv(_file_path, sep="\t", header=False, index=False, encoding="utf-8")
+        
+        file_path = study_path / (
+            ts_file_type.value
+            if not (area_id or constraint_id or group_id)
```
Comment (MartinBelthle) [inline src/antares/service/local_services/area_local.py:346]: i don't see a group_id inside the class TimeSeriesFileType, i think you should remove it. And you shouldn't expect a time_series inside a read method

File: tests/antares/services/local_services/test_area.py [hunk 4]
```diff
@@ -1092,3 +1091,39 @@ def test_read_areas_local(self, local_study_w_areas):
...
+
+    def test_read_load_empty_local(self, local_study_w_areas):
+        study_path = local_study_w_areas.service.config.study_path
+        local_study_object = read_study_local(study_path)
+        areas = local_study_object.read_areas()
+
+        for area in areas:
+            expected_conversion = pd.DataFrame([])
+            
+            load_object = area.read_load()
```
Comment (MartinBelthle) [inline tests/antares/services/local_services/test_area.py:1128]: Would have to change as the Load object doesn't exist anymore.

**Review by MartinBelthle — CHANGES_REQUESTED** [review]

File: src/antares/service/local_services/area_local.py (hunk 2 above)
Comment (killian-scalian) [reply src/antares/service/local_services/area_local.py:362]: It is to replace the class TimeSeriesFile which did this

File: src/antares/service/local_services/area_local.py (hunk 3 above)
Comment (killian-scalian) [reply src/antares/service/local_services/area_local.py:346]: Next commit

File: tests/antares/services/local_services/test_area.py (hunk 4 above)
Comment (killian-scalian) [reply tests/antares/services/local_services/test_area.py:1128]: Next commit

File: src/antares/service/local_services/area_local.py [hunk 5]
```diff
@@ -314,10 +316,25 @@ def delete_renewable_clusters(self, area: Area, renewable_clusters: List[Renewab
 
     def delete_st_storages(self, area: Area, storages: List[STStorage]) -> None:
         raise NotImplementedError
+    
+    def _read_timeseries(self, ts_file_type: TimeSeriesFileType, study_path: Path, area_id: Optional[str] = None, constraint_id: Optional[str] = None, time_series: Optional[pd.DataFrame] = None,) -> pd.DataFrame:
```
Comment (MartinBelthle) [inline src/antares/service/local_services/area_local.py:320]: remove time_series from this call as you shouldn't have to give TS when reading TS

File: src/antares/service/local_services/area_local.py [hunk 6]
```diff
@@ -314,10 +316,25 @@ def delete_renewable_clusters(self, area: Area, renewable_clusters: List[Renewab
...
-        raise NotImplementedError
+        if os.path.getsize(file_path) != 0:
+            _time_series = df_read(file_path)
+        else:
+            _time_series = pd.DataFrame()
 
+        return _time_series
+   
+    def get_load_matrix(self, area: Area) -> pd.DataFrame:
+        timeseries = self._read_timeseries(TimeSeriesFileType.LOAD, self.config.study_path, area_id=area.id)
```
Comment (MartinBelthle) [inline src/antares/service/local_services/area_local.py:335]: You can just return in one line

File: tests/antares/services/local_services/test_area.py [hunk 7]
```diff
@@ -1115,3 +1114,46 @@ def test_read_areas_local(self, local_study_w_areas):
...
+        areas = local_study_object.read_areas()
+
+        def _write_file(_file_path, _time_series) -> None:
+            _file_path.parent.mkdir(parents=True, exist_ok=True)
+            _time_series.to_csv(file_path, sep="\t", header=False, index=False, encoding="utf-8")
+        
+        for area in areas:
+            expected_conversion = pd.DataFrame([])
+            
+            file_path = study_path / "input" / "load" / "series" / f"load_{area.id}.txt".format(area_id=area.id)
```
Comment (MartinBelthle) [inline tests/antares/services/local_services/test_area.py:1156]: You already have an f-string, remove the .format(...)

File: tests/antares/services/local_services/test_area.py [hunk 8]
```diff
@@ -1115,3 +1114,46 @@ def test_read_areas_local(self, local_study_w_areas):
...
+        study_path = local_study_w_areas.service.config.study_path
+        local_study_object = read_study_local(study_path)
+        areas = local_study_object.read_areas()
+
+        def _write_file(_file_path, _time_series) -> None:
+            _file_path.parent.mkdir(parents=True, exist_ok=True)
+            _time_series.to_csv(file_path, sep="\t", header=False, index=False, encoding="utf-8")
+
+        for area in areas:
+            expected_conversion = pd.DataFrame(
```
Comment (MartinBelthle) [inline tests/antares/services/local_services/test_area.py:1130]: You don't expect conversion but classic TS. Doesn't really matter but it's misleading to create such a weird df when not needed. Also you could perhaps merge the 2 tests as it's a lot of duplicated code

**Review by MartinBelthle — CHANGES_REQUESTED** [review]
Still minor issues

File: src/antares/service/local_services/area_local.py (hunk 6 above)
Comment (killian-scalian) [reply src/antares/service/local_services/area_local.py:335]: Next commit

File: tests/antares/services/local_services/test_area.py (hunk 8 above)
Comment (killian-scalian) [reply tests/antares/services/local_services/test_area.py:1130]: Next commit

**Review by MartinBelthle — APPROVED** [review]

**Pull request merged**